prov/rxm/src/rxm_ep.c

```diff
@@ -1597,8 +1597,8 @@ static int rxm_ep_msg_cq_open(struct rxm_ep *rxm_ep, enum fi_wait_obj wait_obj)
 
 	assert((wait_obj == FI_WAIT_NONE) || (wait_obj == FI_WAIT_FD));
 
-	cq_attr.size = (rxm_ep->rxm_info->tx_attr->size +
-			rxm_ep->rxm_info->rx_attr->size);
+	cq_attr.size = (rxm_ep->msg_info->tx_attr->size +
```

Comment:
If it controls by FI_UNIVERSE_SIZE, it can lead to unlimited growth of cq_attr.size.
For example, we can set FI_UNIVERSE_SIZE=2048 (to fix limit of utility AV = 1024) for runs with 2048 MPI ranks, it will lead that cq_attr.size = (1024+ 1024) * 2048. Can the value be accepted by all drivers (HFI1, MLX4/5, I40IW, etc)? perhaps, we need to keep the value less than upper bound accepted by the driver.

Reply:
Yes, it's in the hands of the user to set it appropriately so they don't exceed the device maximum / system memory.

The CQ size is based on the MSG EP tx and rx size which is what is needed as a single RxM endpoint can get universe size * MSG EP rx context size worth of messages for example. The MSG EP tx and rx sizes are by default set to 128 each. So in case of FI_UNIVERE_SIZE=2048, the CQ size would be (128 + 128) * 2048. I believe there isn't a way to get the max CQ count size in OFI yet. May be once we have that we can enforce an upper limit.